aim/web/ui/src/pages/RunDetail/RunLogsTab/RunLogsTab.tsx

```diff
@@ -0,0 +1,187 @@
+
+import LogRow from './LogRow';
+import { IRunLogsTabProps, LogsLastRequestEnum } from './RunLogsTab.d';
+
+import './RunLogsTab.scss';
+
+const SINGLE_LINE_HEIGHT = 16.5;
+const LOAD_MORE_LOGS_COUNT = 100;
+
+function RunLogsTab({
+  isRunLogsLoading,
+  runHash,
+  runLogs,
+  inProgress,
+  updatedLogsCount,
+}: IRunLogsTabProps) {
+  const liveUpdate = React.useRef<any>(null);
+  const logsContainerRef = React.useRef<any>(null);
+  const listRef = React.useRef<any>({});
+  const runsBatchRequestRef = React.useRef<any>({});
+  const [lastRequestType, setLastRequestType] =
+    React.useState<LogsLastRequestEnum>(LogsLastRequestEnum.DEFAULT);
+  const rangeRef = React.useRef<any>(null);
+  const dataRef = React.useRef<any>(null);
+  const [keysList, setKeyList] = React.useState<any>(null);
+  const visibleItemsRange = React.useRef<any>(null);
+
+  React.useEffect(() => {
+    runsBatchRequestRef.current = runDetailAppModel.getRunLogs({ runHash });
+    runsBatchRequestRef.current.call();
+
+    analytics.pageView(ANALYTICS_EVENT_KEYS.runDetails.tabs.logs.tabView);
+    startLiveUpdate();
+
+    return () => {
+      runsBatchRequestRef.current.abort();
+      if (liveUpdate.current?.intervalToken) {
+        clearInterval(liveUpdate.current.intervalToken);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  function liveUpdateCallBack() {
+    return function () {
+      setLastRequestType(LogsLastRequestEnum.LIVE_UPDATE);
+      runsBatchRequestRef.current = runDetailAppModel.getRunLogs({
+        runHash,
+        record_range: dataRef.current
+          ? `${+rangeRef.current?.[1] > 5 ? +rangeRef.current?.[1] - 5 : 0}:`
+          : '',
+        isLiveUpdate: true,
+      });
+      runsBatchRequestRef.current.call();
+    };
+  }
+
+  function startLiveUpdate() {
+    if (inProgress) {
+      const intervalToken = setInterval(liveUpdateCallBack(), 3000);
+      liveUpdate.current = {
+        intervalToken,
+      };
+    }
+  }
+
+  function onScroll(props: ListOnScrollProps) {
+    if (
+      props.scrollOffset === 0 &&
+      +keysList[0] !== 0 &&
+      props.scrollDirection === 'backward'
+    ) {
+      if (liveUpdate.current?.intervalToken) {
+        clearInterval(liveUpdate.current.intervalToken);
+        runsBatchRequestRef.current.abort();
+      }
+      setLastRequestType(LogsLastRequestEnum.LOAD_MORE);
+      runsBatchRequestRef.current = runDetailAppModel.getRunLogs({
+        runHash,
+        record_range: `${
+          +rangeRef.current?.[0] > LOAD_MORE_LOGS_COUNT
+            ? +rangeRef.current?.[0] - LOAD_MORE_LOGS_COUNT
+            : 0
+        }:${rangeRef.current?.[0]}`,
+        isLiveUpdate: true,
+      });
+      runsBatchRequestRef.current.call();
+      startLiveUpdate();
+    }
+  }
+
+  React.useEffect(() => {
+    if (!inProgress && liveUpdate.current?.intervalToken) {
+      clearInterval(liveUpdate.current.intervalToken);
```

Comment:
whats going on when live update already sent and `inProgress` becomes false?, pls note live update call already sent

Reply:
Currently, we can detect if the is the run finished or not yet. We have to refresh the page for that.